arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/CircuitBreaker.kt

```diff
@@ -158,9 +161,9 @@ class CircuitBreaker constructor(
           }
           is ExitCase.Failure -> {
             // Failed reset, which means we go back in the Open state with new expiry val nextTimeout
-            val value = (resetTimeout.millis * exponentialBackoffFactor).toLong().milliseconds
-            val nextTimeout =
-              if (/*maxResetTimeout.isFinite &&*/ value.nanoseconds > maxResetTimeout.nanoseconds) maxResetTimeout
+            val value: Duration = (resetTimeout * exponentialBackoffFactor)
+            val nextTimeout: Duration =
+              if (/*maxResetTimeout.isFinite &&*/ value > maxResetTimeout) maxResetTimeout
```

Comment:
```suggestion
              if (maxResetTimeout.isFinite() && value > maxResetTimeout) maxResetTimeout
```

This is available on `kotlin.time.Duration` and is why I left it commented out ;)